overlord/hookstate/ctlcmd/set.go

```diff
@@ -20,18 +20,28 @@
 package ctlcmd
 
 import (
+	"encoding/json"
 	"fmt"
+	"strings"
 
 	"github.com/snapcore/snapd/i18n"
 )
 
 type setCommand struct {
 	baseCommand
+
+	Positional struct {
+		ConfValues []string `positional-arg-name:"<conf value>" description:"configuration value (key=value)" required:"1"`
+	} `positional-args:"yes" required:"yes"`
 }
 
 var shortSetHelp = i18n.G("Set snap configuration")
 var longSetHelp = i18n.G(`
-The set command is currently only a placeholder.`)
+The set command sets configuration parameters for the snap determined via the
+SNAP_CONTEXT environment variable. This command accepts a number of key=value
```

Comment:
SNAP_CONTEXT Is an implementation detail. We set it, and we get it. People shouldn't have to know it exists. Suggestion:
```
The set command changes the provided configuration options as requested. For example:

$ snapctl set username=joe password=$PASSWORD

All configuration changes are persisted at once, and only after the hook returns
successfully.